### ISSUE TITLE
Cannot set "Fragment size" lower than 512 bytes in Audio settings

### ISSUE THREAD

**Issue by ale-79** [stella-emu/stella#379]
As title says, if I try to set Fragment size with a value lower than 512 bytes, the change has no effect, as confirmed by the output to the console and the Audio dialog itself when I reopen it. 512 bytes and higher values work. 
If I choose the preset "Ultra quality, minimal lag", then fragment size is correctly set to 128 bytes.

**Comment by ale-79** [comment]
> if I try to set Fragment size with a value lower than 512 bytes, the change has no effect ... 

Correction: it **does** have the effect of setting it to 512 bytes.
(e.g. if it was 1 KB and I try to set it to 128 bytes, it change to 512 bytes instead)

**Comment by sa666666** [comment]
Confirmed.   Looking into it now.

**Comment by sa666666** [comment]
I will have to get @DirtyHairy to look at this one, since while I've found the error, I'm not sure why the code is written the way it is, and hence don't want to change it.

It happens in `AudioSettings::fragmentSize()`
```
  updatePresetFromSettings();
  return customSettings() ? convertInt(mySettings.getInt(SETTING_FRAGMENT_SIZE), DEFAULT_FRAGMENT_SIZE) : myPresetFragmentSize;
}

The call to `convertInt` is passing in '128' as the value to use, and '512' as the default.  However, that function is doing something very weird:
```  
uInt32 convertInt(int x, int defaultValue)
{
  return x <= defaultValue ? defaultValue : x;
}

So since 128 < 512, 512 is used.  So I don't see how any value < 512 can ever be used.  This is obviously a mistake, but I'm not sure what the method is supposed to be doing, so I defer to @DirtyHairy for this.

**Comment by DirtyHairy** [comment]
I agree, that's obviously slightly bananas :smirk: I'll look into it later.

**Comment by DirtyHairy** [comment]
Fixed in 1a4e9560
